src/mlpack/core/cv/metrics/facilities.hpp

```diff
@@ -26,6 +26,8 @@ namespace cv {
   * @param labels Labels.
   * @param callerDescription A description of the caller that can be used for
   *     error generation.
+  * @deprecated Check the new versions, util::CheckSameSizes &
+  * util::CheckSameDimensionality.
```

Comment:
Honestly I think it would be fine to remove these functions since we are not using them anywhere else (I think).  This isn't a user-facing function or a function I would expect any user code to be using, so it should be okay to simply remove it. :+1:

Reply:
Okay, cool!!

Reply:
@rcurtin, I have pushed the recent changes as you suggested. All the instances of `AssertSizes` have been updated with the recent proposed apis.  And I think it's better to adapt the whole codebase with these apis in a separate issue & PR (maybe it may help the newcomers to get familiarized with the codebase).  Btw, thanks for those awesome reviews :)